packages/flutter/lib/src/cupertino/tab_view.dart

```diff
@@ -187,6 +187,10 @@ class _CupertinoTabViewState extends State<CupertinoTabView> {
   Widget build(BuildContext context) {
     final Widget child = Navigator(
       key: _navigatorKey,
+      // Manually use a NavigatorPopHandler to handle backs only on the active
+      // tab, instead of letting Navigator use its default back handling for
+      // nested Navigators.
+      handlesBacksWhenNested: false,
```

Comment:
This is unfortunate. if there are multiple parallel navigators, what decide which one receives the pop?

Reply:
See this test for how tabbed navigation is typically done here: https://github.com/flutter/flutter/blob/8d3d3ee4d078301fed75a1d6b984f0280a592fe4/packages/flutter/test/cupertino/tab_scaffold_test.dart#L1274

Each tab has its own nested Navigator wrapped by NavigatorPopHandler. Only the single active tab's NavigatorPopHandler is enabled though, so it's the only one that receives a call to onPop.

Or for the more generic case, where you have 2 sibling Navigators each wrapped with a PopScope, and both nested under a root Navigator? Both will be called. I think that's probably what we want. https://github.com/flutter/flutter/blob/8d3d3ee4d078301fed75a1d6b984f0280a592fe4/packages/flutter/lib/src/widgets/routes.dart#L1788

But for the even more generic case where you have two WidgetsBindingObservers listening for didPopRoute, only the one that was registered first will be called. I think that's very arbitrary and probably a mistake. It would probably solve a lot of the complexity with PopScope etc. if it was able to just call it on the current route or something.

https://github.com/flutter/flutter/blob/8d3d3ee4d078301fed75a1d6b984f0280a592fe4/packages/flutter/lib/src/widgets/binding.dart#L859

Reply:
@chunhtai I wonder what you think about this handlesBacksWhenNested parameter versus scrapping this PR and adding NavigatorPopHandler to go_router?